pkg/client/k8s.go

```diff
@@ -493,3 +481,44 @@ func (k K8sClient) GetBatchJobs() (*batchV1.JobList, error) {
 func (k K8sClient) DeleteBatchJob(name string, namespace string) error {
 	return k.K8sCS.BatchV1().Jobs(namespace).Delete(context.TODO(), name, metav1.DeleteOptions{})
 }
+
+/*
+	LOCAL VOLUMES SPECIFIC METHODS
+*/
+
+// GetDeploymentList returns the deployment-list with a specific
+// label selector query
+func (k K8sClient) GetDeploymentList(labelSelector string) (*appsv1.DeploymentList, error) {
+	if pv, err := k.K8sCS.AppsV1().Deployments("").List(context.TODO(), metav1.ListOptions{
+		LabelSelector: labelSelector,
+	}); err == nil && len(pv.Items) >= 1 {
+		return pv, nil
+	}
+	return nil, fmt.Errorf("got 0 deployments with label-Selector as %s", labelSelector)
+}
+
+// GetNodes returns a list of nodes with the name of nodes
+func (k K8sClient) GetNodes(nodes []string, label, field string) (*corev1.NodeList, error) {
+	// 1. Get all nodes
+	n, err := k.K8sCS.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{
+		LabelSelector: label,
+		FieldSelector: field})
+	if len(nodes) == 0 {
+		return n, err
+	}
+	// 2. Put them in a map[string]*corev1.Node
```

Comment:
```suggestion
	// 2. Put them in a map[string]corev1.Node
```

Reply:
I'll put them in a map of string -> pointer to node instead, I wish there was an easy way to do this without resorting to unstructured objects(the java dev inside me misses generics).